executor/src/witgen/data_structures/finalizable_data.rs

```diff
@@ -146,39 +318,41 @@ impl<T: FieldElement> FinalizableData<T> {
     pub fn take_transposed(&mut self) -> impl Iterator<Item = (PolyID, (Vec<T>, BitVec))> {
         log::debug!(
             "Transposing {} rows with {} columns...",
-            self.data.len(),
+            self.len(),
             self.column_ids.len()
         );
         log::debug!("Finalizing remaining rows...");
-        let mut counter = 0;
-        for i in 0..self.data.len() {
-            if self.finalize(i) {
-                counter += 1;
-            }
-        }
-        log::debug!("Needed to finalize {} / {} rows.", counter, self.data.len());
+        let counter = self.finalize_range(
+            (self.pre_finalized_data.len() + self.finalized_data.len())..self.len(),
+        );
+        log::debug!("Needed to finalize {} / {} rows.", counter, self.len());
```

Comment:
this is also different from before right? there was no concept of pre/post and everything was finalized here?

Reply:
I even considered removing this call to `finalize`. The reason is that we copy out the data anyway, so it doesn't really matter if it has been finalized or not.